pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -730,31 +730,38 @@ func (c *Cluster) removeMonWithOptionalQuorum(daemonName string, shouldRemoveFro
 	}
 	logger.Infof("ensuring removal of unhealthy monitor %s", daemonName)
 
-	resourceName := resourceName(daemonName)
```

Comment:
The changes in this file are really just to re-order the resource cleanup with the mon removal.